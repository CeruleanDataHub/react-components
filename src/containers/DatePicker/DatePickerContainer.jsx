import PropTypes from "prop-types";
import React, { forwardRef } from "react";
import styled from "styled-components";

const Container = styled.button`
  display: flex;
  align-items: center;
  width: 300px;
  height: 40px;
  border: 1px solid #0f181c;
  background: #0f181c;
  padding: 1.25em;
  color: #999;
  font-weight: bold;
  border-radius: 4px;
  cursor: pointer;
`;

export const DatePickerContainer = forwardRef(({ value, onClick }, ref) => (
  <Container onClick={onClick} ref={ref}>
    {value}
  </Container>
));

DatePickerContainer.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func
};

DatePickerContainer.defaultProps = {
  value: "",
  onClick: () => null
};
