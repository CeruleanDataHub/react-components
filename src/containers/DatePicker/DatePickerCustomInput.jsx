import PropTypes from "prop-types";
import React, { forwardRef } from "react";
import styled from "styled-components";

import { Icon } from "../Icon/Icon";

const Container = styled.button`
  display: flex;
  align-items: center;
  width: 300px;
  height: 40px;
  border: 1px solid #0f181c;
  background: #0f181c;
  padding: 1.25em;
  color: #999;
  font-weight: bold;
  border-radius: 4px;
  cursor: pointer;
`;

const CalendarIcon = styled.span`
  font-size: 1.5em;
  margin-right: 1em;
`;

export const DatePickerCustomInput = forwardRef(({ value, onClick }, ref) => (
  <Container onClick={onClick} ref={ref}>
    <Icon name="calendar" as={CalendarIcon} />
    {value}
  </Container>
));

DatePickerCustomInput.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func
};

DatePickerCustomInput.defaultProps = {
  value: "",
  onClick: () => null
};
