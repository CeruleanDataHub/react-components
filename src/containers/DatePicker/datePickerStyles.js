import { css } from "styled-components";

const defaultDatePickerStyles = css`
  .DateInput {
    color: #999;

    &_fangStroke {
      fill: #0f181c;
    }

    &_input {
      color: inherit;
      background-color: #0f181c;
      border-radius: 0 4px 4px 0;

      &__focused {
        border-bottom-color: transparent;
      }
    }
  }

  .DayPicker {
    box-shadow: none;
    background: #0f181c;
    border-radius: 0;
    &_transitionContainer {
      border-radius: 0;
      background: #0f181c;
    }
  }

  .CalendarMonthGrid {
    background: #0f181c;
  }

  .CalendarMonth {
    background: #0f181c;
    padding: 0 13px 10px 13px;
    &_caption {
      color: #999;
      padding-bottom: 42px;
    }
  }

  .DayPickerNavigation_button {
    &__default {
      background: transparent;
      border: none;
    }
  }

  .CalendarDay {
    &__default {
      background: #0f181c;
      color: #999;
      border: 1px solid black;
      &:hover {
        background: white;
        color: black;
      }
    }

    &__selected,
    &__selected_span {
      background: #90ee7e;
      color: black;
    }
  }
`;

export const SingleDatePickerStyles = css`
  .SingleDatePicker {
    color: #999;
    cursor: pointer;

    .SingleDatePickerInput {
      background: #0f181c;
      border-radius: 4px;
      border: none;
    }
    ${defaultDatePickerStyles};
  }
`;

export const DateRangePickerStyles = css`
  .DateRangePicker {
    color: #999;
    cursor: pointer;

    .DateRangePickerInput {
      background: #0f181c;
      border-radius: 4px;
      border: none;
    }
  }
  ${defaultDatePickerStyles};
`;
