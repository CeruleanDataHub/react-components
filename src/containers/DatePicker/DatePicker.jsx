import "react-datepicker/dist/react-datepicker.css";

import enGb from "date-fns/locale/en-GB";
import PropTypes from "prop-types";
import React, { useState } from "react";
import DatePickerWrapper, { registerLocale } from "react-datepicker";
import styled from "styled-components";

import { DatePickerCustomInput } from "./DatePickerCustomInput";
import { DatePickerStyles } from "./datePickerStyles";

registerLocale("en-gb", enGb);

const Container = styled.div`
  ${DatePickerStyles}
`;

export const DatePicker = ({ dateFormat, monthsShown }) => {
  const [startDate, setStartDate] = useState(new Date());
  const ref = React.createRef();

  return (
    <Container>
      <DatePickerWrapper
        selected={startDate}
        onChange={date => setStartDate(date)}
        dateFormat={dateFormat}
        formatWeekDay={day => day.substring(0, 1)}
        customInput={<DatePickerCustomInput ref={ref} />}
        locale="en-gb"
        monthsShown={monthsShown}
      />
    </Container>
  );
};

DatePicker.propTypes = {
  /** How the date should be displayed */
  dateFormat: PropTypes.string,
  /** How many months are shown */
  monthsShown: PropTypes.number
};

DatePicker.defaultProps = {
  dateFormat: "dd.MM.yyyy",
  monthsShown: 1
};
