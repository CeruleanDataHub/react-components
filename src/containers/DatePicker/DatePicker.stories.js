import React from "react";

import { DatePicker } from "./DatePicker";

export default {
  title: "DatePicker",
  component: DatePicker
};

export const DatePickerStory = () => (
  <div style={{ minHeight: 400, width: 300 }}>
    <DatePicker />
  </div>
);

export const DatePickerCustomDateFormatStory = () => (
  <div style={{ minHeight: 400, width: 300 }}>
    <DatePicker dateFormat="MM/dd/yyyy" />
  </div>
);

export const DatePickerTwoMonthsShownStory = () => (
  <div style={{ minHeight: 400, width: 300 }}>
    <DatePicker monthsShown={2} />
  </div>
);
