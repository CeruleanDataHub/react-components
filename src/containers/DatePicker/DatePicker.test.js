import React from "react";
import renderer from "react-test-renderer";

import { DatePicker } from "./DatePicker";

describe("DatePicker", () => {
  it("should render", () => {
    const component = renderer.create(<DatePicker />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render DatePicker with custom dateFormat", () => {
    const component = renderer.create(<DatePicker dateFormat="MM/dd/yyyy" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
