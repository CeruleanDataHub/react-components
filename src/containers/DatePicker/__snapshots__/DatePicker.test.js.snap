// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render 1`] = `
.c4 {
  font-size: 4rem;
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  color: black;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid #0f181c;
  background: #0f181c;
  padding: 1.25em;
  color: #999;
  font-weight: bold;
  border-radius: 4px;
  cursor: pointer;
}

.c3 {
  font-size: 1.5em;
  margin-right: 1em;
}

.c0 .react-datepicker {
  border-color: black;
  background: #0f181c;
  padding: 1em;
}

.c0 .react-datepicker-wrapper {
  width: 100%;
}

.c0 .react-datepicker__header {
  background: #0f181c;
  border-bottom: none;
}

.c0 .react-datepicker__navigation {
  top: 40px;
}

.c0 .react-datepicker__navigation--previous {
  left: 15px;
}

.c0 .react-datepicker__navigation--next {
  right: 15px;
}

.c0 .react-datepicker__triangle {
  display: none;
}

.c0 .react-datepicker__current-month {
  color: #999;
  margin: 0;
  border: 1px solid black;
  padding: 0.25rem;
  height: 40px;
  line-height: 40px;
}

.c0 .react-datepicker__month {
  margin: 0;
}

.c0 .react-datepicker__week {
  border-bottom: 1px solid black;
  border-left: 1px solid black;
}

.c0 .react-datepicker__day-names {
  border-bottom: 1px solid black;
}

.c0 .react-datepicker__day-name {
  color: #999;
  margin: 0;
  border-right: 1px solid black;
  padding: 0.25rem;
}

.c0 .react-datepicker__day-name:first-child {
  border-left: 1px solid black;
}

.c0 .react-datepicker__day {
  color: #999;
  border-radius: 0;
  border-right: 1px solid black;
  margin: 0;
  padding: 0.25rem;
}

.c0 .react-datepicker__day--selected {
  background: #90ee7e;
  color: black;
}

.c0 .react-datepicker__day--keyboard-selected {
  background: none;
}

.c0 .react-datepicker__day:hover {
  color: black;
}

.c0 .react-datepicker__day--outside-month {
  color: transparent;
  pointer-events: none;
}

<div
  className="c0"
>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <div
        className="c1"
        fontFamily="openSans"
        size="normal"
      >
        <button
          className="c2"
          onClick={[Function]}
        >
          <span
            className="c3 c4 lni lni-calendar"
          />
          29.06.2020
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker should render DatePicker with custom dateFormat 1`] = `
.c4 {
  font-size: 4rem;
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  color: black;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid #0f181c;
  background: #0f181c;
  padding: 1.25em;
  color: #999;
  font-weight: bold;
  border-radius: 4px;
  cursor: pointer;
}

.c3 {
  font-size: 1.5em;
  margin-right: 1em;
}

.c0 .react-datepicker {
  border-color: black;
  background: #0f181c;
  padding: 1em;
}

.c0 .react-datepicker-wrapper {
  width: 100%;
}

.c0 .react-datepicker__header {
  background: #0f181c;
  border-bottom: none;
}

.c0 .react-datepicker__navigation {
  top: 40px;
}

.c0 .react-datepicker__navigation--previous {
  left: 15px;
}

.c0 .react-datepicker__navigation--next {
  right: 15px;
}

.c0 .react-datepicker__triangle {
  display: none;
}

.c0 .react-datepicker__current-month {
  color: #999;
  margin: 0;
  border: 1px solid black;
  padding: 0.25rem;
  height: 40px;
  line-height: 40px;
}

.c0 .react-datepicker__month {
  margin: 0;
}

.c0 .react-datepicker__week {
  border-bottom: 1px solid black;
  border-left: 1px solid black;
}

.c0 .react-datepicker__day-names {
  border-bottom: 1px solid black;
}

.c0 .react-datepicker__day-name {
  color: #999;
  margin: 0;
  border-right: 1px solid black;
  padding: 0.25rem;
}

.c0 .react-datepicker__day-name:first-child {
  border-left: 1px solid black;
}

.c0 .react-datepicker__day {
  color: #999;
  border-radius: 0;
  border-right: 1px solid black;
  margin: 0;
  padding: 0.25rem;
}

.c0 .react-datepicker__day--selected {
  background: #90ee7e;
  color: black;
}

.c0 .react-datepicker__day--keyboard-selected {
  background: none;
}

.c0 .react-datepicker__day:hover {
  color: black;
}

.c0 .react-datepicker__day--outside-month {
  color: transparent;
  pointer-events: none;
}

<div
  className="c0"
>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <div
        className="c1"
        fontFamily="openSans"
        size="normal"
      >
        <button
          className="c2"
          onClick={[Function]}
        >
          <span
            className="c3 c4 lni lni-calendar"
          />
          06/29/2020
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker should render DatePicker with two months 1`] = `
.c4 {
  font-size: 4rem;
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  color: black;
}

.c2 {
  font-family: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  border: 1px solid #0f181c;
  background: #0f181c;
  padding: 1.25em;
  color: #999;
  font-weight: bold;
  border-radius: 4px;
  cursor: pointer;
}

.c3 {
  font-size: 1.5em;
  margin-right: 1em;
}

.c0 .react-datepicker {
  border-color: black;
  background: #0f181c;
  padding: 1em;
}

.c0 .react-datepicker-wrapper {
  width: 100%;
}

.c0 .react-datepicker__header {
  background: #0f181c;
  border-bottom: none;
}

.c0 .react-datepicker__navigation {
  top: 40px;
}

.c0 .react-datepicker__navigation--previous {
  left: 15px;
}

.c0 .react-datepicker__navigation--next {
  right: 15px;
}

.c0 .react-datepicker__triangle {
  display: none;
}

.c0 .react-datepicker__current-month {
  color: #999;
  margin: 0;
  border: 1px solid black;
  padding: 0.25rem;
  height: 40px;
  line-height: 40px;
}

.c0 .react-datepicker__month {
  margin: 0;
}

.c0 .react-datepicker__week {
  border-bottom: 1px solid black;
  border-left: 1px solid black;
}

.c0 .react-datepicker__day-names {
  border-bottom: 1px solid black;
}

.c0 .react-datepicker__day-name {
  color: #999;
  margin: 0;
  border-right: 1px solid black;
  padding: 0.25rem;
}

.c0 .react-datepicker__day-name:first-child {
  border-left: 1px solid black;
}

.c0 .react-datepicker__day {
  color: #999;
  border-radius: 0;
  border-right: 1px solid black;
  margin: 0;
  padding: 0.25rem;
}

.c0 .react-datepicker__day--selected {
  background: #90ee7e;
  color: black;
}

.c0 .react-datepicker__day--keyboard-selected {
  background: none;
}

.c0 .react-datepicker__day:hover {
  color: black;
}

.c0 .react-datepicker__day--outside-month {
  color: transparent;
  pointer-events: none;
}

<div
  className="c0"
>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <div
        className="c1"
        fontFamily="openSans"
        size="normal"
      >
        <button
          className="c2"
          onClick={[Function]}
        >
          <span
            className="c3 c4 lni lni-calendar"
          />
          29.06.2020
        </button>
      </div>
    </div>
  </div>
</div>
`;
